src/libpriv/rpmostree-core.c

```diff
@@ -3772,20 +3772,21 @@ rpmostree_context_get_tmprootfs_dfd  (RpmOstreeContext *self)
   return self->tmprootfs_dfd;
 }
 
-static gboolean
-pkg_name_is_kernel (const char *name)
-{
-  const char *kernel_names[] = {"kernel", "kernel-core", "kernel-rt", "kernel-rt-core", NULL};
-  return g_strv_contains (kernel_names, name);
-}
-
-/* Determine if a txn element contains vmlinuz.  TODO: Check via provides?
+/* Determine if a txn element contains vmlinuz via provides.
  * There's also some hacks for this in libdnf.
  */
 static gboolean
 rpmte_is_kernel (rpmte te)
 {
-  return pkg_name_is_kernel (rpmteN (te));
+  const char *kernel_names[] = {"kernel", "kernel-core", "kernel-rt", "kernel-rt-core"};
```

Comment:
```suggestion
  const char *kernel_names[] = {"kernel", "kernel-core", "kernel-rt", "kernel-rt-core", NULL};
```

Reply:
@jlebon Why should we include null in the list if we're checking specifically for kernels?

Reply:
We're passing it to `g_strv_contains` which expects a NULL-terminated array of strings: https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strv-contains.